lib/app/dispatcher.rb

```diff
@@ -55,12 +59,21 @@ def pool
         )
       end
 
-      def scheduler
-        @scheduler ||= Core::NativeScheduler.new(POLL_IDLING)
+      def sync_scheduler
+        @sync_scheduler ||= Core::NativeScheduler.new(:sync, POLL_IDLING)
```

Comment:
I am not sure I like the direction it's taking. This triples the number of pools, therefore increasing the number of threads and consumed memory.

Reply:
Yes, I try to make the tasks isolated.

>This triples the number of pools

It won't triple the number of pools. We will still have one thread pool, which is responsible for task execution of 1) sync job, 2) heartbeat, 3) configuration

Besides the main thread, now we do have 2 more threads to handle heartbeat and configuration (I used `Thread.new` instead of thread pool because they are not meant to be reused). The drawback is we now have 3 threads trying to call `client.indices.search`, but I think this is acceptable. Because the configuration task needs to be run more frequently if we use the same loop it has to run as frequently as the configuration task, overall the number of `client.indices.search` calls will be similar.